cli/mender-artifact/write.go

```diff
@@ -363,12 +364,34 @@ func applySoftwareVersionToTypeInfoProvides(ctx *cli.Context, typeInfoProvides a
 			typeInfoProvides = make(map[string]string)
 		}
 		for key, value := range softwareVersionMapping {
-			typeInfoProvides[key] = value
+			if typeInfoProvides[key] == "" || softwareVersionOverridesProvides(ctx, key) {
+				typeInfoProvides[key] = value
+			}
 		}
 	}
 	return typeInfoProvides
 }
 
+func softwareVersionOverridesProvides(ctx *cli.Context, key string) bool {
+	cmdLine := strings.Join(os.Args, " ")
+
+	var providesVersion string = `(-p|--provides)(\s+|=)` + regexp.QuoteMeta(key) + ":"
```

Comment:
Darn, I was hoping there would be a more straightforward way in urfave/cli.

Reply:
I couldn't find a way, the position of the flags is not exposed.